basilisk/src/ast/interpreter/overload.h

```
@@ -0,0 +1,543 @@
+
+static
+void interpreter_reset_scalar (scalar s)
+{
+  Intergrid * p = (Intergrid *) grid;
+  char * c = p->d + s.i*sizeof (real);
+  memset (c, 0, sizeof (real));
+  char * flags = c + sizeof(real) - 1;
+  *flags = unset;
+}
+
+double z_indexing (scalar s, bool leaves)
```

Clang (14.0.6)

[warning] 2 adjacent parameters of 'z_indexing' of similar type ('int') are easily swapped by mistake

(bugprone-easily-swappable-parameters)